huaweicloud/services/kafka/resource_huaweicloud_dms_kafka_volume_auto_expand_configuration.go

```diff
@@ -0,0 +1,141 @@
+package kafka
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+var volumeAutoExpandConfigurationNonUpdatableParams = []string{
+	"instance_id",
+	"auto_volume_expand_enable",
+	"expand_threshold",
+	"expand_increment",
+	"max_volume_size",
+}
+
+// @API Kafka PUT /v2/{project_id}/instances/{instance_id}/auto-volume-expand
+func ResourceVolumeAutoExpandConfiguration() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVolumeAutoExpandConfigurationCreate,
+		ReadContext:   resourceVolumeAutoExpandConfigurationRead,
+		UpdateContext: resourceVolumeAutoExpandConfigurationUpdate,
+		DeleteContext: resourceVolumeAutoExpandConfigurationDelete,
+
+		CustomizeDiff: config.FlexibleForceNew(volumeAutoExpandConfigurationNonUpdatableParams),
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `The region where the Kafka instance is located to be configured volume auto-expansion.`,
+			},
+			"instance_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The ID of the Kafka instance.`,
+			},
+			"auto_volume_expand_enable": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: `Whether to enable disk auto-expansion.`,
+			},
+			"expand_threshold": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: `The threshold for triggering disk auto-expansion, in percentage (%).`,
+			},
+			"expand_increment": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: `The percentage of storage space to be expanded out of the total instance storage space, in percentage (%)..`,
```

Comment:
```suggestion
				Description: `The percentage of storage space to be expanded out of the total instance storage space, in percentage (%).`,
```
An extra dot character